addons/website/models/website_page_properties.py

```diff
@@ -201,4 +210,9 @@ def write(self, vals):
                         )
                     record.old_url = new_url
 
+        # Set homepage
+        for record in self:
+            if record.is_homepage:
+                record.website_id.homepage_url = record.url
```

Comment:
I am not sure: why was it not needed before? :thinking:

Reply:
This is part of the fix: before, the `homepage_url` was updated when the user re-checked the Is Homepage checkbox (`_inverse_is_homepage`). Now, we need this in case we only change the page `url`, because we don't call `_inverse_is_homepage` in `_onchange_url`.